metaflow/mflog/__init__.py

```diff
@@ -67,19 +72,47 @@ def update_delay(secs_since_start):
     sigmoid = 1. / (1. + math.exp(-0.01 * secs_since_start + 9.))
     return MIN_UPDATE_DELAY + sigmoid * MAX_UPDATE_DELAY
 
+
+def delayed_update_while(
+    condition,
+    update_fn,
+):
+    # Call update_fn() while condition_fn() returns True, with sigmoid delay between
+    # calls.
+    start_time = time.time()
+    next_update_time = start_time
+    next_update_delay = 1
+
+    while True:
+        if time.time() > next_update_time:
+            update_fn()
+            now = time.time()
+            next_update_delay = update_delay(now - start_time)
+            next_update_time = now + next_update_delay
+
+        if not condition():
+            return
+
+        # We should exit this loop when the condition() returns False without
+        # a long delay, regardless of the delay schedule
+        d = min(next_update_delay, 5.0)
+        select.poll().poll(d * 1000)
```

Comment:
Why not a simple sleep?

Reply:
Ha, [I had the same question](https://github.com/Netflix/metaflow/pull/449#discussion_r608166466), @savingoyal told me but I forgot. Workaround for something on OSX?